### ISSUE TITLE
Issues with segmentation results

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1634]
Segmentation fails in the following datasets under: `duke:/sct_testing/large/`:
All: please add tricky cases and list them below for Charley:

State of SCT: commit 60b9b7c366521f852da20ae17a9bd8f10022d32e:

### t2

- zurich_inspired_004/t2

### t2s

- zurich_inspired_004/t2s
- sct_testing_data/mt/mt1.nii.gz
- sct_testing_data/unf_sct_019/t2s/t2s.nii.gz

### dwi

- zurich_inspired_004/dmri
- sct_testing_data/unf_sct_019/dmri/dwi_moco_mean.nii.gz

**Comment by jcohenadad** [comment]
@stephaniealley @fperdigon please add cases as discussed.

**Comment by charleygros** [comment]
Indeed, they are more than welcome!

**Comment by stephaniealley** [comment]
Yes, I will add the ones I've found.

**Comment by stephaniealley** [comment]
@charleygros here is the error I mentioned:

Spinal Cord Toolbox (feature_tf_debian/2145d391e48aac7019dac1cb12db631ff0dbcd06)
Running /home/niftyweb_sct/sct/scripts/sct_deepseg_sc.py -i 9408-0_0-scan1.nii -c t2s -qc ./qc
Using TensorFlow backend.
Loading models...
2018-04-16 16:05:18.143138: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2018-04-16 16:05:18.143152: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2018-04-16 16:05:18.143155: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2018-04-16 16:05:18.143157: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2018-04-16 16:05:18.143159: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
Creating temporary folder...
/tmp/sct-180416160518-udgilh
Reorient the image to RPI, if necessary...
Resample the image to 0.5 mm isotropic resolution...
/home/niftyweb_sct/sct/python/lib/python2.7/site-packages/nipy/io/files.py:145: FutureWarning: Default `strict` currently False; this will change to True in a future version of nipy
  ni_img = nipy2nifti(img, data_dtype = io_dtype)
Finding the spinal cord centerline...
Cropping the image around the spinal cord...
Traceback (most recent call last):
  File "/home/niftyweb_sct/sct/scripts/sct_deepseg_sc.py", line 407, in <module>
    main()
  File "/home/niftyweb_sct/sct/scripts/sct_deepseg_sc.py", line 243, in main
    remove_temp_files=remove_temp_files, verbose=verbose, args=args)
  File "/home/niftyweb_sct/sct/scripts/sct_deepseg_sc.py", line 307, in deep_segmentation_spinalcord
    x_dim_half=crop_size // 2, y_dim_half=crop_size // 2)
  File "/home/niftyweb_sct/sct/scripts/sct_deepseg_sc.py", line 115, in crop_image_around_centerline
    x_ctr, y_ctr = center_of_mass(im_ctr.data[:, :, zz])
  File "/home/niftyweb_sct/sct/python/lib/python2.7/site-packages/scipy/ndimage/measurements.py", line 1289, in center_of_mass
    return [tuple(v) for v in numpy.array(results).T]
TypeError: 'numpy.float64' object is not iterable

**Comment by zougloub** [comment]
Closed by #1706

**Comment by jcohenadad** [comment]
this issue should not have been closed, because its original purpose was to address issues in segmentation results (e.g., two blobs per slice). In the meantime, another error was reported (bug with nii.gz, which by the way, should have been reported in another issue), and the error was then closed after #1706 (specific to the nii.gz issue) has been merged.

again, @fperdigon and @stephaniealley please test this function and add/refer to problematic data under duke:sct_testing [there](https://github.com/neuropoly/spinalcordtoolbox/issues/1634#issue-303313496).
